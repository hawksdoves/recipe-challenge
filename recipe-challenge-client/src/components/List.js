import React from 'react';
import { Link } from 'react-router-dom';
import { Card } from 'semantic-ui-react';

import Recipe from './Recipe';
import Favourite from './Favourite';

function List(props) {
    const { recipes, favourites, ...rest } = props;

    return recipes.length > 0 ?
        (<Card.Group >
            { recipes.map(recipe => {
                return (  
                    <div key={recipe.id} className='recipe' >
                        <Link to={`/recipe/${recipe.id}`} key={`Link-${recipe.id}`} className='card'>
                            <Recipe recipe={recipe} />
                        </Link>
                        <Favourite favourite={favourites.includes(recipe.id)} {...rest} recipeId={recipe.id} />
                    </div>
                )
            })}
        </Card.Group> ) :
        <div>Sorry, nothing matched your filter term</div>
};

export default List;
