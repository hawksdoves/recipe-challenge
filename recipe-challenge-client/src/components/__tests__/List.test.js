import React from 'react';
import { shallow } from 'enzyme';

import List from '../List';

describe('List', () => {
    it('should render all the recipes correctly', () =>{
        const recipes = [
            {
                id: '1',
                ingredients: [{
                    name: 'steak',
                    quantity: '3'
                }]
            },
            {
                id: '2',
                ingredients: [{
                    name: 'chicken',
                    quantity: '2'
                }]
            }
        ];

        const favourites = [];
        const wrapper = shallow(<List recipes={recipes} favourites={favourites} />);

        expect(wrapper).toMatchSnapshot();
    })

    it('should display message if there are no recipes', () =>{
        const recipes = [];
        const favourites = [];
        const wrapper = shallow(<List recipes={recipes} favourites={favourites} />);

        expect(wrapper).toMatchSnapshot();
    })
})
