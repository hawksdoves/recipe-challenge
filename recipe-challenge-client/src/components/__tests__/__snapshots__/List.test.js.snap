// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List should display message if there are no recipes 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    Sorry, nothing matched your filter term
</div>,
  "nodes": Array [
    <div>
      Sorry, nothing matched your filter term
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <List
        favourites={Array []}
        recipes={Array []}
/>,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "favourites": Array [],
          "recipes": Array [],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          Sorry, nothing matched your filter term
</div>,
        "_debugID": 4,
        "_renderedOutput": <div>
          Sorry, nothing matched your filter term
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <List
    favourites={Array []}
    recipes={Array []}
/>,
}
`;

exports[`List should render all the recipes correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <CardGroup>
    <div
        className="recipe"
    >
        <Link
            className="card"
            replace={false}
            to="/recipe/1"
        >
            <Recipe
                recipe={
                    Object {
                        "id": "1",
                        "ingredients": Array [
                          Object {
                            "name": "steak",
                            "quantity": "3",
                          },
                        ],
                      }
                }
            />
        </Link>
        <Favourite
            favourite={false}
            recipeId="1"
        />
    </div>
    <div
        className="recipe"
    >
        <Link
            className="card"
            replace={false}
            to="/recipe/2"
        >
            <Recipe
                recipe={
                    Object {
                        "id": "2",
                        "ingredients": Array [
                          Object {
                            "name": "chicken",
                            "quantity": "2",
                          },
                        ],
                      }
                }
            />
        </Link>
        <Favourite
            favourite={false}
            recipeId="2"
        />
    </div>
</CardGroup>,
  "nodes": Array [
    <CardGroup>
      <div
            className="recipe"
      >
            <Link
                  className="card"
                  replace={false}
                  to="/recipe/1"
            >
                  <Recipe
                        recipe={
                              Object {
                                    "id": "1",
                                    "ingredients": Array [
                                      Object {
                                        "name": "steak",
                                        "quantity": "3",
                                      },
                                    ],
                                  }
                        }
                  />
            </Link>
            <Favourite
                  favourite={false}
                  recipeId="1"
            />
      </div>
      <div
            className="recipe"
      >
            <Link
                  className="card"
                  replace={false}
                  to="/recipe/2"
            >
                  <Recipe
                        recipe={
                              Object {
                                    "id": "2",
                                    "ingredients": Array [
                                      Object {
                                        "name": "chicken",
                                        "quantity": "2",
                                      },
                                    ],
                                  }
                        }
                  />
            </Link>
            <Favourite
                  favourite={false}
                  recipeId="2"
            />
      </div>
</CardGroup>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <List
        favourites={Array []}
        recipes={
                Array [
                        Object {
                          "id": "1",
                          "ingredients": Array [
                            Object {
                              "name": "steak",
                              "quantity": "3",
                            },
                          ],
                        },
                        Object {
                          "id": "2",
                          "ingredients": Array [
                            Object {
                              "name": "chicken",
                              "quantity": "2",
                            },
                          ],
                        },
                      ]
        }
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "favourites": Array [],
          "recipes": Array [
            Object {
              "id": "1",
              "ingredients": Array [
                Object {
                  "name": "steak",
                  "quantity": "3",
                },
              ],
            },
            Object {
              "id": "2",
              "ingredients": Array [
                Object {
                  "name": "chicken",
                  "quantity": "2",
                },
              ],
            },
          ],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <CardGroup>
          <div
                    className="recipe"
          >
                    <Link
                              className="card"
                              replace={false}
                              to="/recipe/1"
                    >
                              <Recipe
                                        recipe={
                                                  Object {
                                                            "id": "1",
                                                            "ingredients": Array [
                                                              Object {
                                                                "name": "steak",
                                                                "quantity": "3",
                                                              },
                                                            ],
                                                          }
                                        }
                              />
                    </Link>
                    <Favourite
                              favourite={false}
                              recipeId="1"
                    />
          </div>
          <div
                    className="recipe"
          >
                    <Link
                              className="card"
                              replace={false}
                              to="/recipe/2"
                    >
                              <Recipe
                                        recipe={
                                                  Object {
                                                            "id": "2",
                                                            "ingredients": Array [
                                                              Object {
                                                                "name": "chicken",
                                                                "quantity": "2",
                                                              },
                                                            ],
                                                          }
                                        }
                              />
                    </Link>
                    <Favourite
                              favourite={false}
                              recipeId="2"
                    />
          </div>
</CardGroup>,
        "_debugID": 2,
        "_renderedOutput": <CardGroup>
          <div
                    className="recipe"
          >
                    <Link
                              className="card"
                              replace={false}
                              to="/recipe/1"
                    >
                              <Recipe
                                        recipe={
                                                  Object {
                                                            "id": "1",
                                                            "ingredients": Array [
                                                              Object {
                                                                "name": "steak",
                                                                "quantity": "3",
                                                              },
                                                            ],
                                                          }
                                        }
                              />
                    </Link>
                    <Favourite
                              favourite={false}
                              recipeId="1"
                    />
          </div>
          <div
                    className="recipe"
          >
                    <Link
                              className="card"
                              replace={false}
                              to="/recipe/2"
                    >
                              <Recipe
                                        recipe={
                                                  Object {
                                                            "id": "2",
                                                            "ingredients": Array [
                                                              Object {
                                                                "name": "chicken",
                                                                "quantity": "2",
                                                              },
                                                            ],
                                                          }
                                        }
                              />
                    </Link>
                    <Favourite
                              favourite={false}
                              recipeId="2"
                    />
          </div>
</CardGroup>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <List
    favourites={Array []}
    recipes={
        Array [
            Object {
              "id": "1",
              "ingredients": Array [
                Object {
                  "name": "steak",
                  "quantity": "3",
                },
              ],
            },
            Object {
              "id": "2",
              "ingredients": Array [
                Object {
                  "name": "chicken",
                  "quantity": "2",
                },
              ],
            },
          ]
    }
/>,
}
`;
